synapse/models/planning.py

```diff
@@ -0,0 +1,155 @@
+                        'doc': 'The title of the phase.'}),
+
+                    ('summary', ('str', {}), {
+                        'doc': 'A summary of the definition of the phase.'}),
+
+                    ('index', ('int', {}), {
+                        'doc': 'The index of this phase within the phases of the system.'}),
+
+                    ('url', ('inet:url', {}), {
+                        'doc': 'A URL which links to the full documentation about the phase.'}),
+
+                    ('system', ('plan:system', {}), {
+                        'doc': 'The planning system which defines this phase.'}),
+                )),
+                ('plan:procedure:type:taxonomy', {}, ()),
+                ('plan:procedure', {}, (
+
+                    ('title', ('str', {}), {
+                        'ex': 'Network Reconnaissance Procedure',
+                        'doc': 'The name of the procedure.'}),
+
+                    ('summary', ('str', {}), {
+                        'doc': 'A summary of the purpose and use cases for the procedure.'}),
+
+                    ('author', ('ps:contact', {}), {
+                        'doc': 'The contact of the person or organization which authored the procedure.'}),
+
+                    ('created', ('time', {}), {
+                        'doc': 'The time the procedure was created.'}),
+
+                    ('updated', ('time', {}), {
+                        'doc': 'The time the procedure was last updated.'}),
+
+                    ('version', ('it:semver', {}), {
+                        'doc': 'The version of the procedure.'}),
+
+                    ('system', ('plan:system', {}), {
+                        'doc': 'The planning system which defines this procedure.'}),
+
+                    ('type', ('plan:procedure:type:taxonomy', {}), {
+                        'doc': 'A type classification for the procedure.'}),
+
+                    ('inputs', ('array', {'type': 'plan:procedure:variable', 'uniq': True, 'sorted': True}), {
+                        'doc': 'An array of inputs required to execute the procedure.'}),
+
+                    ('firststep', ('plan:procedure:step', {}), {
+                        'doc': 'The first step in the procedure.'}),
+                )),
+                ('plan:procedure:variable', {}, (
+
+                    ('name', ('str', {}), {
+                        'doc': 'The name of the variable.'}),
+
+                    ('type', ('str', {}), {
+                        'doc': 'The type for the input. Types are specific to the planning system.'}),
+
+                    ('default', ('data', {}), {
+                        'doc': 'The optional default value if the procedure is invoked without the input.'}),
+
+                    ('procedure', ('plan:procedure', {}), {
+                        'doc': 'The procedure which defines the variable.'}),
+                )),
+                ('plan:procedure:step', {}, (
```

Comment:
Does this need an index as well to account for ordering?

Reply:
No, their order is dictated by the decision tree starting at `plan:procedure:firststep` and is not a linear progression that could be represented by a number.